torch/_functorch/aot_autograd.py

```diff
@@ -1289,6 +1315,11 @@ def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig, *
 
     with context(), track_graph_compiling(aot_config, "inference"):
         compiler = aot_config.inference_compiler if aot_config.inference_compiler is not None else aot_config.fw_compiler
+        if config.functionalize_rng_ops:
+            # Add the seed and offset as example inputs to pass to the compiler
+            fake_mode = get_live_fake_mode_from_dispatch()
+            seed, offset = RNGStateHelper.get_torch_state_as_tuple(fake_mode)
+            flat_args = (seed, offset, *flat_args)
```

Comment:
I probably need to stare some more, but does this mean that all of our compiled graphs will now take in two extra arguments? I'm surprised that didn't break the expecttests in test_aotdispatch.py.

Do you think it would make sense to try to detect if we actually used any randomness in the graph, and avoid passing in the extra seed/offset args if we see that the graph was randomness-free?

Reply:
Currently, all of my work is hidden behind a flag `functorch.config.functionalize_rng_ops`. We add these two new input args only when this flag is turned on.

We could do a  pre-tracing (like we do for metadata) to check if there are any randomness in the graph and avoid additional args. But, we are not at that stage yet.